system/core/pages.php

```diff
@@ -94,139 +94,132 @@ function newPage() {
 		Pulls the language from the url.
 
 		Parameters:
-		args - array with arguments from the page url, below the hypha root url.
-
-		Returns the same array, with any language removed
+		HyphaRequest $hyphaRequest
 	*/
-	function loadLanguage($args) {
-		global $isoLangList, $hyphaLanguage;
+	function loadLanguage(HyphaRequest $hyphaRequest) {
+		global $hyphaLanguage;
+
+		$language = $hyphaRequest->getLanguage();
+
 		// set wiki language. we want to store this in a session variable, so we don't loose language when an image or the settingspage are requested
-		if (count($args) > 0 && array_key_exists($args[0], $isoLangList)) {
-			$hyphaLanguage = $args[0];
-			array_shift($args);
-		} else {
-			$hyphaLanguage = hypha_getDefaultLanguage();
-		}
+        $hyphaLanguage = $language !== null ? $language : hypha_getDefaultLanguage();
```

Comment:
It looks like there is an indentation error here.

Reply:
There was an indentation error. :-)